esp-hal/src/gpio/mod.rs

```diff
@@ -674,13 +684,18 @@ impl<'d> Io<'d> {
     /// Panics if passed interrupt handler is invalid (e.g. has priority
     /// `None`)
     #[instability::unstable]
-    pub fn set_interrupt_handler(&mut self, handler: InterruptHandler) {
-        for core in crate::system::Cpu::other() {
-            crate::interrupt::disable(core, Interrupt::GPIO);
+    pub fn set_interrupt_handler(&mut self, _handler: InterruptHandler) {
+        #[cfg(feature = "rt")]
```

Comment:
This will for sure cause issues. IMHO just silently ignoring function calls because they work with the "other" esp-hal in tree will be ... not a popular choice. I get that you don't really have another option, but right now this is just extra code and indentation for no reason anyway.